# HG changeset patch
# User Emily Chen <emchennyc@gmail.com>
# Date 1357396763 18000
# Node ID af0f1570723397884adb89e31dbcba3cb486a1a9
# Parent  5149624d99e0a7de3dd1d1465092a4a842a30e16
Trac #5956 Graphics object show() doc explains figsize dimension is inches; added doctest for max figsize and figsize values that cause segfault

diff --git a/sage/plot/graphics.py b/sage/plot/graphics.py
--- a/sage/plot/graphics.py
+++ b/sage/plot/graphics.py
@@ -11,6 +11,7 @@
 
 - Jeroen Demeyer (2012-04-19): split off this file from plot.py (:trac:`12857`)
 - Punarbasu Purkayastha (2012-05-20): Add logarithmic scale (:trac:`4529`)
+- Emily Chen (2013-01-05): Add documentation for :meth:`~sage.plot.graphics.Graphics.show` figsize parameter (:trac:`5956`)
 
 """
 
@@ -1272,9 +1273,12 @@
 
         - ``filename`` - (default: None) string
 
-        - ``dpi`` - dots per inch
-
-        - ``figsize`` - [width, height]
+        - ``dpi`` - (default: 100) dots per inch
+
+        - ``figsize`` - (default: [8.0,6.0]) [width, height] inches. The
+          maximum value of each of the width and the height can be 327
+          inches, at the default ``dpi`` of 100 dpi, which is just shy of
+          the maximum allowed value of 32768 dots per inch.
 
         - ``fig_tight`` - (default: True) whether to clip the drawing
           tightly around drawn objects.  If True, then the resulting
@@ -1455,13 +1459,16 @@
               your own tick formatter using the option ``tick_formatter``.
 
         EXAMPLES::
-        
+
             sage: c = circle((1,1), 1, color='red')
             sage: c.show(xmin=-1, xmax=3, ymin=-1, ymax=3)
-        
-        You could also just make the picture larger by changing ``figsize``::
-
-            sage: c.show(figsize=8, xmin=-1, xmax=3, ymin=-1, ymax=3)
+
+        You can make the picture larger by changing ``figsize`` with width,
+        height each having a maximum value of 327 inches at default dpi::
+
+            sage: p = ellipse((0,0),4,1)
+            sage: p.show(figsize=[327,10],dpi=100)
+            sage: p.show(figsize=[328,10],dpi=80)
 
         You can turn off the drawing of the axes::
 
@@ -1750,6 +1757,35 @@
             ...
             ValueError: 'title_pos' must be a list or tuple of two real numbers.
 
+        TESTS:
+
+        The figsize width and height parameters must be less than 328
+        inches each, corresponding to the maximum allowed dpi of 32768.::
+
+            sage: p = ellipse((0,0),4,1)
+            sage: p.show(figsize=[328,10],dpi=100)
+            Traceback (most recent call last):
+            ...
+            ValueError: width and height must each be below 32768
+
+        The following test results in a segmentation fault and should not
+        be doctested.::
+
+            sage: p = ellipse((0,0),4,1)
+            sage: p.show(figsize=[232,232],dpi=100) # not tested
+            Unhandled SIGSEGV: A segmentation fault occurred in Sage.
+            This probably occurred because a *compiled* component of Sage
+            has a bug in it and is not properly wrapped with sig_on(),
+            sig_off(). You might want to run Sage under gdb with 'sage
+            -gdb' to debug this.  Sage will now terminate.
+
+            sage: p.show(figsize=[327,181],dpi=100) # not tested
+            Unhandled SIGSEGV: A segmentation fault occurred in Sage.
+            This probably occurred because a *compiled* component of Sage
+            has a bug in it and is not properly wrapped with sig_on(),
+            sig_off(). You might want to run Sage under gdb with 'sage
+            -gdb' to debug this.  Sage will now terminate.
+
         """
 
         # This option should not be passed on to save().
